Revit_Engine/Create/Elements/Roof.cs

```diff
@@ -0,0 +1,63 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2018, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System.ComponentModel;
+
+using BH.oM.Architecture.Elements;
+using BH.oM.Reflection.Attributes;
+using BH.oM.Common.Properties;
+using BH.oM.Geometry;
+
+namespace BH.Engine.Adapters.Revit
+{
+    public static partial class Create
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Creates Roof object.")]
+        [Input("object2DProperties", "Object2DProperties")]
+        [Input("edges", "External edges of Roof")]
+        [Output("Roof")]
+        public static Roof Roof(Object2DProperties object2DProperties, ICurve edges)
```

Comment:
Same as for floor. No methods like this in Revit_Engine

Reply:
I fully agree on that however can we first fully agree on IObject2D, Wall etc. structure (methods, properties etc.) before we spread the changes across to different repositories? I assume this is proof of concept for now.